compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/platform/UIKitTextInputService.uikit.kt

```diff
@@ -395,6 +412,14 @@ internal class UIKitTextInputService(
             floatingCursorTranslation = null
         }
 
+        override fun beginEditBatch() {
+            editBatchDepth++
+        }
+
+        override fun endEditBatch() {
+            editBatchDepth--
```

Comment:
Force flush added

Reply:
`updateState` calls on every value change. There is no sense to flush commands buffer on this event.

Reply:
@igordmn please check comment above ^^

Reply:
> updateState calls on every value change

It is called on every recomposition as far as I remember. It can also be triggered not only by the event changing the state, but also by some other callback.

For, example, we clear text in `onClick { text = "" }`.

If we won't flush, there can be a potential race between `updateState("")` and the postponed commands that will add more text to the already cleared value:

```
setMarkedText("1")
setMarkedText("12")
onClick { text = "" }`
updateState("")
flush() -> onValueChanged("12) -> text = "12"
```
The input event happen sequentially before the click event, so if we postpone - we violate the order of text changes.

Reply:
> It is called on every recomposition as far as I remember

On iOS it calls on every value change, also as a result of flushing current pool of commands. 

The problem is that this function calls already after the value is updated, so the sequence mentioned by you won't change the result if we add force flush.
Another problem is that by force-flushing commands in `updateState`, we're applying these flushed commands before the current command is being completed. As the result, we will break the commands order.

Reply:
> The problem is that this function calls already after the value is updated, so the sequence mentioned by you won't change the result if we add force flush.

If the state change already happened on click (text clearead), but we perform postponed edits later, then we still have a bug - the field won't be empty.

Should we do something else then? For example, cancel flushing in case of a new value? Maybe we should also somehow check the passed `oldValue` to write code correctly.

Reply:
Or flush it during every `key`, `pointer`, `input` event to achieve the correct event order.